micrometer-core/src/main/java/io/micrometer/core/instrument/HighCardinalityTagsDetector.java

```diff
@@ -135,44 +137,51 @@ public void close() {
 
     private void detectHighCardinalityTags() {
         try {
-            findFirst().ifPresent(this.meterNameConsumer);
+            findFirst().ifPresent((meterInfo) -> {
+                if (this.meterNameConsumer != null) {
+                    this.meterNameConsumer.accept(meterInfo.getName());
```

Comment:
Rather than keep two consumers, could we just keep one since it is an internal detail. For backward compatibility, if a user is configuring a `Consumer<String>` we can convert that to a `Consumer<HighCardinalityMeterInfo>` and then always run that. I would probably even go so far as to mark the `Consumer<String>` API deprecated.